uctMaths/competition/compadmin_views.py

```diff
@@ -59,6 +64,47 @@ def upload_results(request):
 
     return render_to_response('admin/upload_results.html', c, context_instance=RequestContext(request))
 
+@login_required
+def upload_declaration(request):
+    """ Handle upload files from the user. Bound to the upload_declaration.html admin page."""
+
+    handler_output=[]#Sent to html template for feedback to user
+
+    #Once the user has pressed 'Submit'
+    if request.method == 'POST':
+        #a=request.POST
+        try: #Try receive file from 'Submit' post from user
+            form = UploadDeclarationForm(request.POST, request.FILES)
+            file = request.FILES['upload_file']
+            filepath = os.path.join(__file__,"..","..","Declaration")
+            filename = str(file)
+            if not filename == 'Declaration.pdf':
```

Comment:
why don't we just set `filename` to `Declaration.pdf` and ignore what the user called the file.